### PR TITLE
Add compile details endpoint

### LINKED ISSUE #3132: Create compilerun detail endpoint
Create an endpoint the allows to fetch a complete compile run object. A complete compile run object should also contain the reports associated with the compile run. Currently there is only a v1 endpoint for this.

The goal is to:

1. Create an endpoint `/v2/compilereport/<id>`
2. This endpoint returns the full compile run. This is related to #3131 
3. Add reports to the CompileRun and also return the compile_data that is left out #3131 
4. The report list should be optional and typed with a dto that has the same fields the Report database object:
    - id
    - started
    - completed
    - name
    - errstream
    - outstream
    - returncode

### PR DESCRIPTION
**Pull request by andraskvr**
# Description

closes #3132 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [x] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: src/inmanta/data/__init__.py [hunk 1]
```diff
@@ -2619,13 +2619,121 @@ async def get_compile_reports(
...
+        ) SELECT * FROM compiledetails;
+        """
+        values = [cls._get_value(environment), cls._get_value(id)]
+        result = await cls.select_query(query, values, no_obj=True)
+        result = cast(List[Record], result)
+
+        if not result:
+            return None
+        # Get the details from the requested compile, and the reports from the substitute
+        records = list(filter(lambda r: r["id"] == id, result))
```
Comment (arnaudsjs) [inline src/inmanta/data/__init__.py:2699]: This I don't understand. I think we need to increase the verbosity of the documentation here.

File: src/inmanta/data/__init__.py [hunk 2]
```diff
@@ -2619,13 +2619,121 @@ async def get_compile_reports(
...
+            success=requested_compile["success"],
+            version=requested_compile["version"],
+            do_export=requested_compile["do_export"],
+            force_update=requested_compile["force_update"],
+            metadata=json.loads(requested_compile["metadata"]) if requested_compile["metadata"] else {},
+            environment_variables=json.loads(requested_compile["environment_variables"])
+            if requested_compile["environment_variables"]
+            else {},
+            compile_data=json.loads(requested_compile["compile_data"]) if requested_compile["compile_data"] else None,
+            reports=reports if reports else None,
```
Comment (arnaudsjs) [inline src/inmanta/data/__init__.py:2734]: Why set to `None` instead of an empty list?

File: src/inmanta/data/model.py [hunk 3]
```diff
@@ -157,6 +157,22 @@ class CompileReport(CompileRunBase):
     version: Optional[int]
 
 
+class Report(BaseModel):
```
Comment (arnaudsjs) [inline src/inmanta/data/model.py:160]: I wound make the name more specific than just `Report`. It's very likely that we will add other dto's in the future that will represent some kind of report. That would be confusing.

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: src/inmanta/data/__init__.py (hunk 1 above)
Comment (andraskvr) [reply src/inmanta/data/__init__.py]: I added some more comments, let me know if you think it should be more detailed

**Review by wouterdb — APPROVED** [review]

**Review by arnaudsjs — APPROVED** [review]

**Pull request closed without merging**